api/controllers/DashboardController.js

```diff
@@ -133,177 +238,62 @@ var dashboardController = {
             pageSize,
             req,
             res,
+            results,
         } = displayAppsArgs;
-        //redirect to 404 if user has manually set a page in the query string
+        const { totalPages, paginationMessage } =
+            dashboardController._paginationAndPageTotal(
+                results,
+                offset,
+                pageSize
+            );
         if (results.length === 0) {
             if (currentPage != 1) {
                 return res.view('404.ejs');
             } else {
                 sails.log.error('No results found.');
```

Comment:
what happens to users who see this? are they meant to see a specific page? looks like they get "stuck" here, as opposed to currentPage != 1 users who get a 404 explicitly returned to them